modules/copy-post.php

```diff
@@ -0,0 +1,310 @@
+		add_filter( 'default_title', array( $this, 'filter_title' ), 10, 2 );
+		add_filter( 'default_content', array( $this, 'filter_content' ), 10, 2 );
+		add_filter( 'default_excerpt', array( $this, 'filter_excerpt' ), 10, 2 );
+
+		/**
+		 * Fires after all updates have been performed, and default content filters have been added.
+		 * Allows for any cleanup or post operations, and default content filters can be removed or modified.
+		 *
+		 * @module copy-post
+		 *
+		 * @since 7.0
+		 *
+		 * @param WP_Post $source_post Post object that was copied.
+		 * @param int     $target_post_id Target post ID.
+		 * @param array   $update_results Results of all update operations, allowing action to be taken.
+		 */
+		do_action( 'jetpack_copy_post', $source_post, $target_post_id, $update_results );
+	}
+
+	/**
+	 * Determine if the current user has access to the source post.
+	 *
+	 * @param int $post_id Source post ID (the post being copied).
+	 * @return bool True if user has the meta cap of `read_post` for the given post ID, false otherwise.
+	 */
+	protected function user_can_access_post( $post_id ) {
+		return current_user_can( 'read_post', $post_id );
+	}
+
+	/**
+	 * Update the target post's title, content, excerpt, categories, and tags.
+	 *
+	 * @param WP_Post $source_post Post object to be copied.
+	 * @param int     $target_post_id Target post ID.
+	 * @return int    0 on failure, or the updated post ID on success.
+	 */
+	protected function update_content( $source_post, $target_post_id ) {
+		$data = array(
+			'ID'             => $target_post_id,
+			'post_title'     => $source_post->post_title,
+			'post_content'   => $source_post->post_content,
+			'post_excerpt'   => $source_post->post_excerpt,
+			'comment_status' => $source_post->comment_status,
+			'ping_status'    => $source_post->ping_status,
+			'post_category'  => $source_post->post_category,
+			'tags_input'     => $source_post->tags_input,
+		);
+
+		/**
+		 * Fires just before the target post is updated with its new data.
+		 * Allows for final data adjustments before updating the target post.
+		 *
+		 * @module copy-post
+		 *
+		 * @since 7.0
+		 *
+		 * @param array $data Post data with which to update the target (new) post.
+		 * @param WP_Post $source_post Post object being copied.
+		 * @param int     $target_post_id Target post ID.
+		 */
+		$data = apply_filters( 'jetpack_copy_post_data', $data, $source_post, $target_post_id );
+		return wp_update_post( $data );
+	}
+
+	/**
+	 * Update terms for custom post types.
+	 *
+	 * @param WP_Post $source_post Post object to be copied.
+	 * @param int     $target_post_id Target post ID.
+	 * @return array Results of attempts to set each term to the target (new) post.
+	 */
+	protected function update_custom_post_type_terms( $source_post, $target_post_id ) {
+		$results = array();
+		if ( in_array( $source_post->post_type, array( 'post', 'page' ), true ) ) {
```

Comment:
I would recommend adding a filter there, so plugin / theme authors can add their CPTs to the list of post types to ignore if needed. I could see this being useful when the terms being used are meant to be unique per post.

Reply:
Great idea 👍